lib/extensions/explain_validity.py

```diff
@@ -0,0 +1,154 @@
+            }
+            )
+        layer.append(text)
+
+        tspan = inkex.etree.Element(SVG_TSPAN_TAG)
+        if message == "Self-intersection":
+            tspan.text = _("Self-intersection")
+        elif message == "Hole lies outside shell":
+            tspan.text = _("Gap")
+        else:
+            # different error (message not translatable)
+            tspan.text = message
+        text.append(tspan)
+
+    def get_or_create_validity_layer(self):
+        svg = self.document.getroot()
+        layer = svg.find(".//*[@id='__validity_layer__']")
+
+        if layer is None:
+            layer = inkex.etree.Element(
+                SVG_GROUP_TAG,
+                {
+                    'id': '__validity_layer__',
+                    INKSCAPE_LABEL: _('Explain Validity'),
+                    INKSCAPE_GROUPMODE: 'layer',
+                    SODIPODI_INSENSITIVE: "true"
+                })
+            svg.append(layer)
+            layer = svg.find(".//*[@id='__validity_layer__']")
+
+            ignore_layer = IgnoreValidityLayer()
+            ignore_layer.insert_layer_ignore_command(layer)
+
+            text_x = str(self.unittouu(svg.get('width')) + 5)
+
+            text_container = inkex.etree.Element(
+                SVG_TEXT_TAG,
+                {
+                    "x": text_x,
+                    "y": str(5),
+                    "style": "fill:#000000;font-size:5px;line-height:1;"
+                }
+                )
+            layer.append(text_container)
+
+            text = [
+                        [_("Explain Validity"), "font-weight: bold; font-size: 6px;"],
+                        ["", ""],
+                        [_("Self-Intersection"), "font-weight: bold;"],
+                        [_("* Path > Union (Ctrl++)"), "font-size: 4px;"],
+                        [_("* Path > Break apart (Shift+Ctrl+K)"), "font-size: 4px;"],
+                        ["", ""],
+                        [_("Gap"), "font-weight: bold;"],
+                        [_("* Path > Break apart (Shift+Ctrl+K)"), "font-size: 4px;"],
+                        [_("* (Optional) Recombine shapes with holes (Ctrl+K)."), "font-size: 4px;"],
+                        ["", ""],
+                        [_("It is possible, that one object contains more than one error,"), "font-style: italic; font-size: 3px;"],
+                        [_("yet there will be only one pointer per object."), "font-style: italic; font-size: 3px;"],
+                        [_("Run this function again, when further errors occur."), "font-style: italic; font-size: 3px;"],
+                        ["", ""],
+                        [_('Remove pointers by deleting the layer named "Explain Validity"'), "font-style: italic; font-size: 3px;"],
+                        [_("through the objects panel (Object -> Objects...)."), "font-style: italic; font-size: 3px;"]
+                ]
+
+            for text_line in text:
```

Comment:
That's an elegant loop right there. :smile: 

I'm a little worried this may create a ton of confusing work for translators, since there are so many lines of text that will show up as different translation strings.  On the other hand, I love how the formatted text looks in the SVG -- it's really professional looking.

I wonder if we might want to evolve this extension in the future by moving it into Electron.  We could add a link to documentation about Shapely errors on `inkstitch.org`.  Maybe we could even show each shape to the user separately in the UI and point out the invalid spots.

For now, this works fine as it is, and I really want to get this extension out to our users, so consider this just brainstorming for the future :)